### PR TITLE
CASMPET-5620 Add additional TLS tests

### PR DESCRIPTION
**Pull request by kburns-hpe**
## Summary and Scope

Adds tests to validate that the platform CA is included in the ca-bundle.pem and that a new version of certifi has not been installed.

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves [CASMPET-5620](https://jira-pro.its.hpecorp.net:8443/browse/CASMPET-5620)

### Tested on:

  * Shandy

### Test description:

Validated that goss tests passed on systems setup correctly and failed when modified.

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why? Y
- Was upgrade tested? If not, why? N, new tests.
- Was downgrade tested? If not, why? N, new tests.
- Were new tests (or test issues/Jiras) created for this change? Y

## Pull Request Checklist

- [X] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [X] License file intact
- [X] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Review by rgross727 — APPROVED** [review]

File: goss-testing/tests/ncn/goss-platform-ca-in-bundle.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#
+# MIT License
+#
+# (C) Copyright 2014-2022 Hewlett Packard Enterprise Development LP
```
Comment (mharding-hpe) [inline goss-testing/tests/ncn/goss-platform-ca-in-bundle.yaml:4]:
```suggestion
# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
```

File: goss-testing/tests/ncn/goss-validate-certifi-version.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#
+# MIT License
+#
+# (C) Copyright 2014-2022 Hewlett Packard Enterprise Development LP
```
Comment (mharding-hpe) [inline goss-testing/tests/ncn/goss-validate-certifi-version.yaml:4]:
```suggestion
# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
```

File: goss-testing/tests/ncn/goss-validate-certifi-version.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+command:
```
Comment (mharding-hpe) [inline goss-testing/tests/ncn/goss-validate-certifi-version.yaml:24]: May as well log the output of this test.

```suggestion
{{ $logrun := .Env.GOSS_BASE | printf "%s/scripts/log_run.sh" }}
command:
```

File: goss-testing/tests/ncn/goss-validate-certifi-version.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  validate_certifi_version:
```
Comment (mharding-hpe) [inline goss-testing/tests/ncn/goss-validate-certifi-version.yaml:25]:
```suggestion
  {{ $testlabel := "validate_certifi_version" }}
  {{$testlabel}}:
```

File: goss-testing/tests/ncn/goss-validate-certifi-version.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    title: "Validate certifi is the correct version."
+    meta:
+      desc: "If this fails then validate that an upstream version of certifi has not been installed. If a non-SLES certifi package has been installed the uninstall it by running `pip uninstall certifi`."
+      sev: 0
+    exec: "pip show certifi"
```
Comment (mharding-hpe) [inline goss-testing/tests/ncn/goss-validate-certifi-version.yaml:30]:
```suggestion
    exec: |-
      "{{$logrun}}" -l "{{$testlabel}}" \
        pip show certifi
```

File: goss-testing/tests/ncn/goss-validate-certifi-version.yaml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+command:
+  validate_certifi_version:
+    title: "Validate certifi is the correct version."
+    meta:
+      desc: "If this fails then validate that an upstream version of certifi has not been installed. If a non-SLES certifi package has been installed the uninstall it by running `pip uninstall certifi`."
```
Comment (mharding-hpe) [inline goss-testing/tests/ncn/goss-validate-certifi-version.yaml:28]:
```suggestion
      desc: "If this fails, then validate that an upstream version of certifi has not been installed. If a non-SLES certifi package has been installed, then uninstall it by running `pip uninstall certifi`."
```

**Review by mharding-hpe — CHANGES_REQUESTED** [review]

**Review by rgross727 — APPROVED** [review]

**Review by mharding-hpe — APPROVED** [review]

**Pull request merged**